<?xml version="1.0"?>
<launch>
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find shibuyne_pointcloud)/config/normal_lcl.rviz" /> -->
  <!-- <include file="$(find pointcloud_analyzer)/launch/pointcloud_capturer.launch"/> -->
  <!-- <include file="$(find ransac_ground_filter)/launch/sq_ground_filter.launch"/> -->
  <!-- <include file="$(find velodyne_height_map)/launch/shibuya_height.launch"/> -->
	<!--Transform-->
	<node pkg="tf" type="static_transform_publisher" name="centerlaser_transform" args="0 0 0 0 0 0 base_link centerlaser 100" />

	<!-- /scan -> /cloud -->
	<!-- /sq_lidar/scan -> /scan_to_cloud -->
  <node name="sq_lidar"  pkg="sq_lidar" type="sq_lidar">
		<remap from="/scan" to="/sq_lidar/scan"/>
		<remap from="/cloud" to="/scan_to_cloud"/>
	</node>

	<!-- /cloud -> /cloud/3d_unflatness -->
  <node name="fast_unflatness_calculator_3d"  pkg="unflatness_evaluator" type="fast_unflatness_calculator_3d" output="screen">
		<!-- subscriber -->
		<remap from="/cloud" to="/scan_to_cloud"/>
		<param name="filter_distance_thre" value="3.5"/>
		<param name="filter_height_thre" value="0.06"/>
	</node>
  

	<!-- /cloud/3d_unflatness -> cloud_tf -->
	<node pkg="shibuyne_pointcloud" type="transform_pointcloud_" name="transform_pointcloud_">
		<remap from="/cloud" to="/cloud/3d_unflatness"/>
		<param name="target_frame" type="string" value="base_link"/>
	</node>

		<!-- cloud_tf -> edge or plane -->
	<node name="devide_unflatness" pkg="unflatness_evaluator" type="devide_unflatness">
		<!-- 小さくするとエッジが増える -->
		<!-- <param name="edge_thre" value="0.5"/> -->
		<param name="edge_thre" value="0.3"/>
	</node>

	<!--lcl-->
	<!-- /cloud_transformed -> /cloud/lcl -->
	<node name="lcl_"  pkg="shibuyne_pointcloud" type="lcl_">
		<!-- <param name="save_num" type="int" value="30"/> -->
		<!-- <param name="save_num" type="int" value="50"/> -->
		<!-- <param name="save_num" type="int" value="80"/> -->
		<param name="save_num" type="int" value="90"/>
		<param name="z_threshold" type="double" value="2.9"/>
		<!-- <param name="z_threshold" type="double" value="50.0"/> -->
		<!-- <param name="z_threshold_max" type="double" value="2.9"/> -->
		<remap from="/cloud" to="/cloud/tf"/>
		<remap from="/odom" to="/odom/complement"/>
		<!-- <param name="save_num_2" type="int" value="100"/> -->
	</node>

</launch>